File: .yalc/@entur/atb-mobile-client-sdk/token/token-state-android-lib/src/main/java/no/entur/abt/android/token/remote/RemoteTokenService.java
package no.entur.abt.android.token.remote;

import no.entur.abt.android.token.ActivatedToken;
import no.entur.abt.android.token.PendingToken;
import no.entur.abt.android.token.Token;
import no.entur.abt.android.token.attestation.DeviceAttestationException;
import no.entur.abt.android.token.exception.DeviceAttestationRemoteException;
import no.entur.abt.android.token.exception.TokenCommandException;
import no.entur.abt.android.token.remote.exception.RemoteTokenStateException;
import no.entur.abt.core.exchange.pb.v1.SecureContainer;

public interface RemoteTokenService<T> {

	ActiveTokenDetails getMobileTokenDetails(Token<T> token, SecureContainer secureContainerToken, T traceId)
			throws DeviceAttestationException, DeviceAttestationRemoteException, RemoteTokenStateException;

	PendingTokenDetails initiateMobileTokenRenewal(ActivatedToken<T> currentToken, SecureContainer secureContainerToken, T traceId)
			throws DeviceAttestationException, DeviceAttestationRemoteException, RemoteTokenStateException;

	ActiveTokenDetails completeMobileTokenRenewal(ActivatedToken<T> currentToken, PendingToken<T> pendingToken, SecureContainer secureContainerToken)
			throws DeviceAttestationException, DeviceAttestationRemoteException, TokenCommandException, RemoteTokenStateException;

	ActiveTokenDetails activateNewMobileToken(PendingToken<T> pendingNewToken) throws DeviceAttestationRemoteException, TokenCommandException;

}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-state-react-native-lib/src/state/remote/RemoteTokenService.ts
import type { ActivatedToken, PendingToken, Token } from '../../../../token-core-javascript-lib'
import type {
    PendingTokenDetails,
    ActiveTokenDetails,
    Reattestation,
} from '../../../../token-server-node-lib/types'

export interface RemoteTokenService {
    initiateNewMobileToken: (
        correlationId: string,
        isEmulator: boolean,
    ) => Promise<PendingTokenDetails>

    activateNewMobileToken: (
        pendingToken: PendingToken,
        correlationId: string,
    ) => Promise<ActiveTokenDetails>

    initiateMobileTokenRenewal: (
        token: Token,
        secureContainer: string,
        correlationId: string,
        reattestation?: Reattestation,
    ) => Promise<PendingTokenDetails>

    completeMobileTokenRenewal: (
        pendingToken: PendingToken,
        secureContainer: string,
        activatedToken: ActivatedToken,
        correlationId: string,
        reattestation?: Reattestation,
    ) => Promise<ActiveTokenDetails>

    getMobileTokenDetails: (
        token: Token,
        secureContainer: string,
        correlationId: string,
        reattestation?: Reattestation,
    ) => Promise<ActiveTokenDetails>
}
